server/game/cards/01_SOR/leaders/CassianAndorDedicatedToTheRebellion.ts

```diff
@@ -0,0 +1,54 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { LeaderUnitCard } from '../../../core/card/LeaderUnitCard';
+import { StateWatcherRegistrar } from '../../../core/stateWatcher/StateWatcherRegistrar';
+import { DamageDealtThisPhaseWatcher } from '../../../stateWatchers/DamageDealtThisPhaseWatcher';
+
+export default class CassionAndorDedicatedToTheRebellion extends LeaderUnitCard {
+    private damageDealtThisPhaseWatcher: DamageDealtThisPhaseWatcher;
+
+    protected override getImplementationId() {
+        return {
+            id: '3187874229',
+            internalName: 'cassian-andor#dedicated-to-the-rebellion',
+        };
+    }
+
+    protected override setupStateWatchers(registrar: StateWatcherRegistrar): void {
+        this.damageDealtThisPhaseWatcher = AbilityHelper.stateWatchers.damageDealtThisPhase(registrar, this);
+    }
+
+    protected override setupLeaderSideAbilities() {
+        this.addActionAbility({
+            title: 'If you\'ve dealt 3 or more damage to an enemy base this phase, draw a card.',
+            cost: [AbilityHelper.costs.abilityResourceCost(1), AbilityHelper.costs.exhaustSelf()],
+            immediateEffect: AbilityHelper.immediateEffects.conditional({
+                condition: (context) => {
+                    const damageDealtToBase = this.damageDealtThisPhaseWatcher.getDamageDealtByPlayer(
+                        context.source.controller,
+                        (damage) => damage.target.isBase() && damage.target.controller !== context.source.controller
```

Comment:
Because the target's controller may have changed since the damage happened (due to e.g. Palpatine ability), we need to use the `damageSource` record for determining any information about the damage. In this case `damageSource.player` indicates the responsible player for the damage

Reply:
Looking at the getDamageDealtByPlayer code, am I right that I didn't need this additional filter at all? It was already filtering by player using damageSource.player?  I should just ditch it?

Reply:
Oh yeah actually that's correct now that you say it, missed that